pkg/investigations/clustermonitoringerrorbudgetburn/clustermonitoringerrorbudgetburn.go

```diff
@@ -0,0 +1,87 @@
+	investigation "github.com/openshift/configuration-anomaly-detection/pkg/investigations"
+	k8sclient "github.com/openshift/configuration-anomaly-detection/pkg/k8s"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	uwmMisconfiguredSL = ocm.ServiceLog{
+		Severity:     "Major",
+		Summary:      "Action required: review user-workload-monitoring configuration",
+		ServiceName:  "SREManualAction",
+		Description:  "Your cluster's user workload monitoring is misconfigured: the user-workload-monitoring-config ConfigMap in the openshift-user-workload-monitoring namespace is misconfigured. Please review your configuration alongside the product documentation: https://access.redhat.com/documentation/en-us/red_hat_openshift_service_on_aws/4/html/monitoring/configuring-the-monitoring-stack#.",
+		InternalOnly: false,
+	}
+)
+
+func Investigate(r *investigation.Resources) error {
+	ctx := context.Background()
+	scheme := runtime.NewScheme()
+
+	// Add config.openshift.io/v1 to scheme for clusteroperator
+	if err := configv1.Install(scheme); err != nil {
+		return fmt.Errorf("unable to add openshift/api/config scheme: %w", err)
+	}
+	// add corev1 to scheme for core k8s
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return fmt.Errorf("unable to add corev1 scheme: %w", err)
+	}
+
+	k8scli, err := k8sclient.New(r.Cluster.ID(), client.Options{Scheme: scheme})
+	if err != nil {
+		return fmt.Errorf("unable to initalize k8s cli: %w", err)
+	}
+
+	// List the monitoring cluster operator
+	coList := &configv1.ClusterOperatorList{}
+	listOptions := &client.ListOptions{FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": "monitoring"})}
+	err = k8scli.List(ctx, coList, listOptions)
+	if err != nil {
+		return fmt.Errorf("unable to list monitoring clusteroperator: %w", err)
+	}
+
+	if len(coList.Items) != 1 {
+		return fmt.Errorf("found %d clusteroperators, expected 1", len(coList.Items))
+	}
+
+	var notesSb strings.Builder
+	notesSb.WriteString("🤖 Automated ClusterMonitoringErrorBudgetBurnSRE pre-investigation 🤖\n")
+	notesSb.WriteString("===========================\n")
+
+	// Check if the status for `Available` corresponds to our expected symptom string
+	monitoringCo := coList.Items[0]
+
+	if statusMatchesSymptom(&monitoringCo) {
+		notesSb.WriteString("🤖 Customer misconfigured the UWM configmap, sending service log and silencing the alert\n")
+		r.OcmClient.PostServiceLog(r.Cluster.ID(), &uwmMisconfiguredSL)
+		return r.PdClient.SilenceAlertWithNote(notesSb.String())
+	}
+
+	notesSb.WriteString("✅ Monitoring CO not degraded due to a broken UWM configmap\n")
+	return r.PdClient.EscalateAlertWithNote(notesSb.String())
+}
+
+// Check if the `Available` status condition reports a broken UWM config
+func statusMatchesSymptom(monitoringCo *configv1.ClusterOperator) bool {
```

Comment:
Major: Can you give this function a less generic name that makes it clear what it's checking? Something like `isUWMConfigInvalid` or so.

Reply:
Good point, thanks!